File: 0001-two-sum/0001-two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        int[] arr=new int[2];
        Map<Integer,Integer> maps=new HashMap<>();
        for(int i=0;i<nums.length;i++){
            if(maps.containsKey(target-nums[i])){
                arr[1]=i;
                arr[0]=maps.get(target-nums[i]);
                return arr;
                
            }
            maps.put(nums[i],i);
        }
        return arr;
    }
}

File: 0001-two-sum/0001-two-sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        //Brute Force
        for(int i=0;i<nums.size();i++){
            int ele=target-nums[i];
            for(int j=i+1;j<nums.size();j++){
                if(ele==nums[j]){
                    return {i,j};
                }
            }
        }
        return {0};
        
        
        //Find best solution == ?
        
        
        //Map solution is also exist
    }
};